inlang/source-code/versioned-interfaces/message-bundle/src/interface.ts

```diff
@@ -0,0 +1,115 @@
+import { LanguageTag as Locale } from "@inlang/language-tag"
+import { Type, type Static } from "@sinclair/typebox"
+
+export type Literal = Static<typeof Literal>
+export const Literal = Type.Object({
+	type: Type.Literal("literal"),
+	value: Type.String(),
+})
+
+/**
+ * A (text) element that is translatable and rendered to the UI.
+ */
+export type Text = Static<typeof Text>
+export const Text = Type.Object({
+	type: Type.Literal("text"),
+	value: Type.String(),
+})
```

Comment:
MessageFormat 2.0 seems to have removed the `Text` type in favor of a pure string. What was their reasoning? 

Notes from me: 

- iterating through pattern elements seems much easier if every node has `node.type` property instead of using union matching e.g. `if (typeof node === string || node.type === XYZ)`